File: pipeline-test/src/main/java/com/pipe/pipeapp/PusherParams.java
package com.pipe.pipeapp;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;
import org.testng.annotations.*;

public final class PusherParams {
    String recFile="";
    String authState = "";
    int optionalPid = 0;// 26955;
    Boolean optionalMobile = false;
    int subsId = 0;//223;
    String domain = "";
    Boolean retry = true;
    String referrer = "";
    String dnsName = "";//"https://wrlinqa-aws.clicktale.net/ctn_v2/";
    String ua = "";
    Boolean isProtocolV16 = false;
    int messageCounter = 0;
    int sessionStreamsDelayFromAuth = 0;
    String optionalSuffix = "";
   // final static public Properties prop = new Properties();


    //In order to run other env please change you testng.xml to the desired env in edit configurations

    public PusherParams(){

         InputStream input = null;
        try{

            recFile =  Configuration.prop.getProperty("suite.pusher.defaultParams.recFile");
            authState = Configuration.prop.getProperty("suite.pusher.defaultParams.authState");
            optionalPid = Integer.parseInt(Configuration.prop.getProperty("suite.pusher.defaultParams.optionalPid"));
            optionalMobile = Boolean.parseBoolean(Configuration.prop.getProperty("suite.pusher.defaultParams.optionalMobile"));
            subsId = Integer.parseInt(Configuration.prop.getProperty("suite.pusher.defaultParams.subsId"));
            domain = Configuration.prop.getProperty("suite.pusher.defaultParams.domain");
            retry = Boolean.parseBoolean(Configuration.prop.getProperty("suite.pusher.defaultParams.retry"));
            referrer = Configuration.prop.getProperty("suite.pusher.defaultParams.referrer");
            dnsName = Configuration.prop.getProperty("suite.pusher.defaultParams.dnsName");
            ua = Configuration.prop.getProperty("suite.pusher.defaultParams.ua");
            isProtocolV16 = Boolean.getBoolean(Configuration.prop.getProperty("suite.pusher.defaultParams.isProtocolV16"));
            messageCounter = Integer.parseInt(Configuration.prop.getProperty("suite.pusher.defaultParams.messageCounter"));
            sessionStreamsDelayFromAuth = Integer.parseInt(Configuration.prop.getProperty("suite.pusher.defaultParams.sessionStreamsDelayFromAuth"));
            optionalSuffix = Configuration.prop.getProperty("suite.pusher.defaultParams.optionalSuffix");
            System.out.println(domain);
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            if (input != null) {
                try {
                    input.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }


    }

    public String getAuthState(){
        return authState;
    }


}

File: Regressions/src/main/scala/com/clicktale/pipeline/dataObjects/PusherParams.scala
package com.clicktale.pipeline.dataObjects

import com.clicktale.pipeline.framework.dal.ConfigParser.conf

case class PusherParams(var recFile:String = "/recFiles/recEvent.json",
                        var authState:String = conf.getString("WebRecorder.Session.States.Recording"),
                        var optionalPid : Int = 26955,
                        var optionalMobile : Boolean = false,
                        var subsId : Int = 223,
                        var domain: String = conf.getString(s"WebRecorder.Environments.${conf.getString("WebRecorder.Current.Environment")}.Domains"),
                        var retry:Boolean = true,
                        var referrer:String = "",
                        var dnsName: String = "https://wrlinqa-aws.clicktale.net/ctn_v2/",
                        var ua: String = "",
                        var isProtocolV16: Boolean = false,
                        var messageCounter:Int = 0,
                        var sessionStreamsDelayFromAuth: Int = 0,
                        var optionalSuffix: String = "") {

  def getRecFile(): String ={
    return recFile
  }
}

